client/src/app/shared/form-card/form-card.component.ts

```diff
@@ -0,0 +1,204 @@
+import { Component, Input } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AccountService } from 'src/app/account-card/services/account.service';
+import { CategoryService } from 'src/app/category/services/category.service';
+import { CurrencyService } from 'src/app/currencies/currency.service';
+import { ReloadService } from 'src/app/reload/reload.service';
+import { TransactionService } from 'src/app/transaction-card/services/transaction.service';
+import { ModalService } from '../modal-card/services/modal-card.service';
+import { NotificationService } from '../notification-card/services/notification.service';
+import { FormCardService } from './services/form-card.service';
+
+@Component({
+  selector: 'app-form-card',
+  templateUrl: './form-card.component.html',
+  styleUrls: ['./form-card.component.scss'],
+})
+export class FormCardComponent {
+  @Input() type: string = '';
+
+  //for account
+  public accountForm: FormGroup = new FormGroup({
+    title: new FormControl(
+      (this.formCardService.isEditing &&
+        this.accountService?.singleAccount?.title) ||
+        '',
+      [Validators.required]
+    ),
+    currency: new FormControl(
+      (this.formCardService.isEditing &&
+        this.accountService?.singleAccount?.currency) ||
+        this.currencyService.defaultCurrency,
+      [Validators.required]
+    ),
+    description: new FormControl(
+      (this.formCardService.isEditing &&
+        this.accountService?.singleAccount?.description) ||
+        ''
+    ),
+  });
+
+  //for transaction
+  public transactionForm: FormGroup = new FormGroup({
+    title: new FormControl(
+      (this.formCardService.isEditing &&
+        this.transactionService?.singleTransaction?.title) ||
+        '',
+      [Validators.required]
+    ),
+    categories: new FormControl(
+      (this.formCardService.isEditing &&
+        this.transactionService?.singleTransaction?.category) ||
+        '',
+      [Validators.required]
+    ),
+    amount: new FormControl(
+      (this.formCardService.isEditing &&
+        this.transactionService?.singleTransaction?.amount) ||
+        '',
+      [Validators.required]
+    ),
+    paymentDate: new FormControl(
+      (this.formCardService.isEditing &&
+        this.transactionService?.singleTransaction?.createdAt) ||
+        '',
+      [Validators.required]
+    ),
+    description: new FormControl(
+      (this.formCardService.isEditing &&
+        this.transactionService?.singleTransaction?.description) ||
+        ''
+    ),
+  });
+  public transactionType: string =
+    (this.formCardService.isEditing &&
+      this.transactionService?.singleTransaction?.type) ||
+    '';
+
+  //for category
+  public categoryType: string = '';
+
+  public categoryForm: FormGroup = new FormGroup({
+    title: new FormControl('', [Validators.required]),
+  });
+
+  public onIncomeClick(): void {
+    if (this.type === 'Category') {
```

Comment:
this.type === 'Category' ? this.categoryType = 'income' : this.transactionType = 'income'